rust/xaynet-server/src/state_machine/phases/unmask.rs

```diff
@@ -39,16 +32,31 @@ impl Phase for PhaseState<Unmask> {
 
     /// Run the unmasking phase
     async fn run(&mut self) -> Result<(), StateError> {
-        metrics!(
-            self.shared.io.metrics_tx,
-            metrics::masks::total_number::update(
-                self.inner.model_mask_dict.len(),
-                self.shared.state.round_id,
-                Self::NAME
-            )
-        );
+        #[cfg(feature = "metrics")]
+        {
+            let redis = &mut self.shared.io.redis;
+            let metrics_tx = &mut self.shared.io.metrics_tx;
+
+            match redis.connection().await.get_number_of_unique_masks().await {
```

Comment:
Could this be `spawned`? I don't think we need to really wait for it.

Reply:
Good idea👍